### PR TITLE
Moving more types from reth crate to node core crate

### PR DESCRIPTION
**Pull request by i-m-aditya**
Ref #6063

### CODE REVIEW

**Comment by i-m-aditya** [comment]
@mattsse Have moves args, and other modules on which it is dependent to the node-core crate. Any other refactoring required?

**Comment by i-m-aditya** [comment]
Will fix the failed checks by EOD. But is this the desired approach.

**Review by mattsse — CHANGES_REQUESTED** [review]
this is a good draft!

we don't want that examples need to also add the node-core crate

instead we want to re-export node-core from the bin/reth crate

**Review by mattsse — CHANGES_REQUESTED** [review]
sweet,

last request to also reexport the types from previous locations, so that this change becomes non breaking. Because this is used by a few users already and we don't want to introduce breaking changes incrementally

File: examples/additional-rpc-namespace-in-cli/src/main.rs [hunk 1]
```diff
@@ -14,12 +14,15 @@
...
 use clap::Parser;
 use jsonrpsee::{core::RpcResult, proc_macros::rpc};
-use reth::cli::{
-    components::{RethNodeComponents, RethRpcComponents},
-    config::RethRpcConfig,
-    ext::{RethCliExt, RethNodeCommandConfig},
-    Cli,
+use reth::{
+    cli::Cli,
+    node_core::cli::{
```
[the remark this answers is not in the document]
Comment (i-m-aditya) [reply examples/additional-rpc-namespace-in-cli/src/main.rs:19]: That would mean having the same modules(args, cli) in both crates: node-core and bin/reth. Should I go ahead and do this change?

[the remark this answers is not in the document]
Comment (mattsse) [reply examples/additional-rpc-namespace-in-cli/src/main.rs:19]: yes please 🙏

adding a few mods with reexports is fine, we want to reduce breaking changes for now

File: examples/trace-transaction-cli/src/main.rs [hunk 2]
```diff
@@ -25,6 +25,7 @@ use reth::{
     tasks::TaskSpawner,
     transaction_pool::TransactionPool,
 };
+
```
Comment (mattsse) [inline examples/trace-transaction-cli/src/main.rs:28]:
```suggestion
```

File: examples/beacon-api-sse/src/main.rs [hunk 3]
```diff
@@ -18,6 +18,9 @@
 use clap::Parser;
 use futures_util::stream::StreamExt;
 use mev_share_sse::{client::EventStream, EventClient};
+use std::net::{IpAddr, Ipv4Addr};
+use tracing::{info, warn};
+
```
Comment (mattsse) [inline examples/beacon-api-sse/src/main.rs:23]:
```suggestion
```

File: examples/additional-rpc-namespace-in-cli/src/main.rs [hunk 4]
```diff
@@ -20,6 +20,7 @@ use reth::cli::{
     ext::{RethCliExt, RethNodeCommandConfig},
     Cli,
 };
+
```
Comment (mattsse) [inline examples/additional-rpc-namespace-in-cli/src/main.rs:23]:
```suggestion
```

File: bin/reth/src/cli/ext.rs [hunk 5]
```diff
@@ -208,7 +208,7 @@ impl<Conf: RethNodeCommandConfig> RethCliExt for NoArgsCliExt<Conf> {
 /// additional CLI arguments.
 ///
 /// Note: This type must be manually filled with a [RethNodeCommandConfig] manually before executing
-/// the [NodeCommand](crate::commands::node::NodeCommand).
+/// the [NodeCommand](https://github.com/paradigmxyz/reth/blob/main/bin/reth/src/commands/node/mod.rs).
```
Comment (Rjected) [inline bin/reth/src/cli/ext.rs:211]: should probably keep them relative to a crate so our doc tests can pick up on any missing links

File: bin/reth/src/commands/init_cmd.rs [hunk 6]
```diff
@@ -1,18 +1,19 @@
 //! Command that initializes the node from a genesis file.
 
+use crate::init::init_genesis;
```
Comment (Rjected) [inline bin/reth/src/commands/init_cmd.rs:3]: nit: this can probably be moved back to the `crate::{` import section

**Review by mattsse — APPROVED** [review]
lgtm, pending @Rjected 

I assume we need a few smaller followups but since this touches a lot we should merge asap to reduce friction

**Review by Rjected — APPROVED** [review]
have some small nits, otherwise LGTM. for followup: removing any code that's now duplicated in `reth_node_core`

**Pull request merged**